runtime/runtime.go

```diff
@@ -385,6 +385,19 @@ func (rt *Runtime) Serve(ctx context.Context) error {
 
 	defer undo()
 
+	// Override the runtime parameters with the manager's router property if set
+	// This takes places before start so that routes can be added in start hook
+	router := rt.Params.Router
+	if r := rt.Manager.GetRouter(); r != nil {
+		router = r
+	}
+
+	// If no router was set by plugin or param, set a default one
+	if router == nil {
+		router = mux.NewRouter()
+		rt.Manager.SetRouter(router)
+	}
+
```

Comment:
Let's refactor this so that the router is created in `NewRuntime(...)` if the `runtime.Params` struct does not contain one already. Let's pass the router (whether it's passed via `runtime.Params` or created in `NewRuntime`) to the `Manager` when it's instantiated. Then let's pass the same router into the server.